libarchive/archive_write_set_format_pax.c

```diff
@@ -1117,15 +1117,17 @@ archive_write_pax_header(struct archive_write *a,
 
 	/*
 	 * Technically, the mtime field in the ustar header can
-	 * support 33 bits, but many platforms use signed 32-bit time
-	 * values.  The cutoff of 0x7fffffff here is a compromise.
+	 * support 36 bits. We are using all of them to keep
+	 * tar/test/test_option_C_mtree.c simple and passing after 2038.
+	 * Platforms that use signed 32-bit time values need to fix
+	 * their handling of timestamps anyway.
 	 * Yes, this check is duplicated just below; this helps to
 	 * avoid writing an mtime attribute just to handle a
 	 * high-resolution timestamp in "restricted pax" mode.
 	 */
 	if (!need_extension &&
 	    ((archive_entry_mtime(entry_main) < 0)
-		|| (archive_entry_mtime(entry_main) >= 0x7fffffff)))
+		|| (archive_entry_mtime(entry_main) >= 0xfffffffff)))
```

Comment:
It's probably worth moving this into a constant so we don't repeat it in multiple places.  Then we can document it where it's defined.

To keep with the standards, we should limit to 33 bits.  That takes us well into the 23rd century, which seems sufficient for the time being.  ;-)